op-chain-ops/genesis/config.go

```diff
@@ -8,76 +8,110 @@ import (
-
-	L2OutputOracleSubmissionInterval uint64         `json:"l2OutputOracleSubmissionInterval"`
-	L2OutputOracleStartingTimestamp  int            `json:"l2OutputOracleStartingTimestamp"`
-	L2OutputOracleProposer           common.Address `json:"l2OutputOracleProposer"`
-	L2OutputOracleOwner              common.Address `json:"l2OutputOracleOwner"`
-	L2OutputOracleGenesisL2Output    common.Hash    `json:"l2OutputOracleGenesisL2Output"`
-
-	SystemConfigOwner common.Address `json:"systemConfigOwner"`
-
-	L1BlockTime                 uint64         `json:"l1BlockTime"`
-	L1GenesisBlockTimestamp     hexutil.Uint64 `json:"l1GenesisBlockTimestamp"`
-	L1GenesisBlockNonce         hexutil.Uint64 `json:"l1GenesisBlockNonce"`
-	CliqueSignerAddress         common.Address `json:"cliqueSignerAddress"` // proof of stake genesis if left zeroed.
-	L1GenesisBlockGasLimit      hexutil.Uint64 `json:"l1GenesisBlockGasLimit"`
-	L1GenesisBlockDifficulty    *hexutil.Big   `json:"l1GenesisBlockDifficulty"`
-	L1GenesisBlockMixHash       common.Hash    `json:"l1GenesisBlockMixHash"`
-	L1GenesisBlockCoinbase      common.Address `json:"l1GenesisBlockCoinbase"`
-	L1GenesisBlockNumber        hexutil.Uint64 `json:"l1GenesisBlockNumber"`
-	L1GenesisBlockGasUsed       hexutil.Uint64 `json:"l1GenesisBlockGasUsed"`
-	L1GenesisBlockParentHash    common.Hash    `json:"l1GenesisBlockParentHash"`
-	L1GenesisBlockBaseFeePerGas *hexutil.Big   `json:"l1GenesisBlockBaseFeePerGas"`
-
-	L2GenesisBlockNonce         hexutil.Uint64 `json:"l2GenesisBlockNonce"`
-	L2GenesisBlockExtraData     hexutil.Bytes  `json:"l2GenesisBlockExtraData"`
-	L2GenesisBlockGasLimit      hexutil.Uint64 `json:"l2GenesisBlockGasLimit"`
-	L2GenesisBlockDifficulty    *hexutil.Big   `json:"l2GenesisBlockDifficulty"`
-	L2GenesisBlockMixHash       common.Hash    `json:"l2GenesisBlockMixHash"`
-	L2GenesisBlockCoinbase      common.Address `json:"l2GenesisBlockCoinbase"`
-	L2GenesisBlockNumber        hexutil.Uint64 `json:"l2GenesisBlockNumber"`
-	L2GenesisBlockGasUsed       hexutil.Uint64 `json:"l2GenesisBlockGasUsed"`
-	L2GenesisBlockParentHash    common.Hash    `json:"l2GenesisBlockParentHash"`
-	L2GenesisBlockBaseFeePerGas *hexutil.Big   `json:"l2GenesisBlockBaseFeePerGas"`
-
-	ProxyAdminOwner             common.Address `json:"proxyAdminOwner"`
-	L2CrossDomainMessengerOwner common.Address `json:"l2CrossDomainMessengerOwner"`
-	OptimismBaseFeeRecipient    common.Address `json:"optimismBaseFeeRecipient"`
-	OptimismL1FeeRecipient      common.Address `json:"optimismL1FeeRecipient"`
-
-	GasPriceOracleOwner    common.Address `json:"gasPriceOracleOwner"`
-	GasPriceOracleOverhead uint64         `json:"gasPriceOracleOverhead"`
-	GasPriceOracleScalar   uint64         `json:"gasPriceOracleScalar"`
-
-	DeploymentWaitConfirmations int `json:"deploymentWaitConfirmations"`
-
-	EIP1559Elasticity  uint64 `json:"eip1559Elasticity"`
-	EIP1559Denominator uint64 `json:"eip1559Denominator"`
+	JSONDeployConfig
```

Comment:
I needed the ability to serialize the `rpc.BlockNumberOrHash` struct to JSON. It's not supported in the Geth library - it outputs an object that then cannot be read. We've run into this issue before and never solved it, so I tried to solve it here without introducing a huge struct with lots of duplicate fields. Open to other solutions, but we need a way to serialize this weird `BlockNumberOrHash` thing to JSON for the migration tests.

Reply:
Perhaps we can open a PR to geth that makes that type json serializable

Reply:
It looks like there is test coverage of this here: https://github.com/ethereum/go-ethereum/blob/d629e0204799f2f7109fd52aa245e852c2e4f921/rpc/types_test.go#L127

Perhaps we are on an older version of geth here?

Reply:
This isn't that bad of a diff to be honest, i think its ok leaving as is

Reply:
I fixed it. Will update this in a moment.